.github/workflows/go-CI.yml

```diff
@@ -0,0 +1,82 @@
+name: Go CI
+
+on:
+  push:
+    branches:
+      - dev
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+
+      - name: Set Up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '1.21.4' 
+
+      # - name: Install golangci-lint
+      #   run: go get -u github.com/golangci/golangci-lint/cmd/golangci-lint
+
+      # - name: Run golangci-lint
+      #   run: golangci-lint run
+
+      - name: Run Tests
+        run: go test -v ./...
+
+      - name: Build
+        run: |
+            echo "Current Directory: $(pwd)"
+            ls -al
+            go build -o myapp
+            ls -al
+
+      - name: 'Upload Goland Build File Artifact whith a retention period 1 day'
```

Comment:
No need, to upload the artificate, the better is to push the built image to docker registry